sheepdog/transactions/deletion/transaction.py

```diff
@@ -87,7 +87,24 @@ def _delete_fields(self):
             for e in self.valid_entities:
                 for field in self.fields_to_delete.split(','):
                     if field in e.node.props:
-                        e.node.props[field] = None
+                        field_is_protected = any(
+                            [field in dictionary.schema[e.node.label].get(protected_category, [])
+                             for protected_category in ['required', 'systemProperties']]
+                        )
+                        # Below is a sad workaround for inconsistent gdcdictionary :|
+                        # Will be safe to remove once 'required' fields is a
+                        # complete and tested set for all node types in gdcdictionary.schemas
+                        if current_app.config.get('IS_GDC', True):
+                            if field in ['submitter_id', 'project_id', 'state', 'file_state']:
```

Comment:
shouldn't we just disallow deleting any systemProperties?

Reply:
This is what I generally do @philloooo, check out the comment above. It is me who added this to the code because sadly it turned out that `required` and `systemProperties` fields in dictionaries are broken (sometimes there are fields that are not there but they are required). I was instructed by @madopal that we can not get dictionary update till next data release and it will require full revision

Reply:
so we are patching gdcdictionary by putting some temporary gdc logic to sheepdog...

Reply:
I know, right? ugly

Reply:
Yeah, the problem is that we don't actually have the required fields in all the cases (like `submitter_id` isn't required everywhere). And therefore if we actually allow them to delete these values...ick.